compiler/src/iree/compiler/Codegen/LLVMGPU/KernelConfig.cpp

```diff
@@ -886,9 +886,25 @@ setAttentionVectorDistributionConfig(IREE::GPU::TargetAttr target,
   SmallVector<NamedAttribute, 2> qkConfig;
   SmallVector<NamedAttribute, 2> pvConfig;
 
+  // Configuring for qk matmul.
+  // subgroup_n count for qk matmul is always 1, because it is the
+  // reduction dimension. The subgroup_n count is in reality, for the pvMatmul.
   IREE::GPU::LoweringConfigAttr::setPromotedOperandList(context, qkConfig,
                                                         {0, 1});
+  IREE::GPU::LoweringConfigAttr::setMmaKind(context, qkConfig,
+                                            mmaAttrs[schedule->index]);
+  IREE::GPU::LoweringConfigAttr::setSubgroupMCount(
+      context, qkConfig, schedule->mSubgroupCounts[0]);
+  IREE::GPU::LoweringConfigAttr::setSubgroupNCount(context, qkConfig, 1);
```

Comment:
Can we write this for docs instead:

The subgroup distribution in attention is controlled by the second matmul (Parallel dimension distribution is usually (almost always) controlled by the last reduction operation in a dispatch). Since VectorDistribution doesn't have logic to set subgroup and thread layouts seperately, we explicitly set the subgroup count for the first matmul as well, corresponding to what the second matmul dictates.

Reply:
Done!